devices/uart-zynq/uart.c

```diff
@@ -460,6 +463,10 @@ static int uart_init(unsigned int minor)
 			* TXEN = 0x1; RXEN = 0x1; TXRES = 0x1; RXRES = 0x1 */
 		*(uart->base + cr) = (*(uart->base + cr) & ~0x000001ff) | 0x00000017;
 	}
+	else {
+		/* Clear RX to discard possible invalid chars from PL when UART is routed via PL */
+		*(uart->base + cr) = *(uart->base + cr) | (1 << 0);
```

Comment:
Regarding the comment - I get the intention, but we're doing it when the UART in not routed via PL too, so it's a little misleading
```suggestion
		/* Clear RX to discard possible invalid chars */
		*(uart->base + cr) |= *(1 << 0);
```

Reply:
sure, now its only executed if UART is routed via PL

Reply:
Yes :), I wanted to avoid if-else "else" part inside conditional macro